src/pkg/services/m365/api/lists.go

```diff
@@ -490,20 +518,57 @@ func setAdditionalDataByColumnNames(
 	fieldData := orig.GetAdditionalData()
 	filteredData := make(map[string]any)
 
-	for colName, colDetails := range columnNames {
-		if val, ok := fieldData[colName]; ok {
+	for _, colDetails := range columnNames {
+		if val, ok := fieldData[colDetails.getFieldName]; ok {
 			setMultipleEnabledByFieldData(val, colDetails)
-
-			filteredData[colName] = fieldData[colName]
+			filteredData[colDetails.createFieldName] = val
+			populateMultipleValues(val, filteredData, colDetails)
 		}
 
-		specifyODataType(filteredData, colDetails, colName)
+		specifyODataType(filteredData, colDetails, colDetails.createFieldName)
 	}
 
 	return filteredData
 }
 
+func populateMultipleValues(val any, filteredData map[string]any, colDetails *columnDetails) {
+	if !colDetails.isMultipleEnabled {
+		return
+	}
+
+	if !colDetails.isPersonColumn {
+		return
+	}
+
+	multiNestedFields, ok := val.([]any)
```

Comment:
You could extend he the `isSlice` check recommended from the other PR here to have a return value like `([]T, bool)`.  That'd be a more centralized way of parsing slices than you have here.  If you do that, I'd sugges moving the func to `common/tform`.